src/main/java/org/broadinstitute/hellbender/tools/exome/allelefraction/AlleleFractionLikelihoods.java

```diff
@@ -0,0 +1,179 @@
+     * <p>
+     * if indicator == REF_MINOR, same as ALT_MINOR but with f <--> 1 - f
+     * <p>
+     * if indicator == OUTLIER log {pi * a!r!/(n+1)!}
+     * <p>
+     * where alpha = mu*beta and n = a + r
+     */
+    public static double hetLogLikelihood(final AlleleFractionState state, final int segment, final AllelicCount count, final AlleleFractionIndicator indicator) {
+        return hetLogLikelihood(state, segment, count, indicator, AllelicPanelOfNormals.EMPTY_PON);
+    }
+
+    public static double hetLogLikelihood(final AlleleFractionState state, final int segment, final AllelicCount count, final AlleleFractionIndicator indicator,
+                                          final AllelicPanelOfNormals allelicPON) {
+        final SimpleInterval site = count.getInterval();
+        final double alpha;
+        final double beta;
+        if (allelicPON.equals(AllelicPanelOfNormals.EMPTY_PON)) {
+            beta = state.meanBias() / state.biasVariance();
+            alpha = state.meanBias() * beta;
+
+        } else {
+            alpha = allelicPON.getAlpha(site);
+            beta = allelicPON.getBeta(site);
+        }
+        final double pi = state.outlierProbability();
+        final double minorFraction = state.minorFractionInSegment(segment);
+        final int a = count.getAltReadCount();
+        final int r = count.getRefReadCount();
+
+        return hetLogLikelihood(alpha, beta, minorFraction, pi, indicator, a, r);
+    }
+
+    public static double hetLogLikelihood(final double alpha, final double beta, final double minorFraction,
+                                          final double outlierProbability, final AlleleFractionIndicator indicator,
+                                          final int a, final int r) {
+        if (indicator == AlleleFractionIndicator.OUTLIER) {
+            return log(outlierProbability) + log10ToLog(log10Factorial(a) + log10Factorial(r) - log10Factorial(a + r + 1));
+        } else {
+            final double f = indicator == AlleleFractionIndicator.ALT_MINOR ? minorFraction : 1 - minorFraction;
+            return log((1 - outlierProbability) / 2) + logPhi(alpha, beta, f, a, r);
+        }
+    }
+
+    protected static double logPhi(final double alpha, final double beta, final double f, final int a, final int r) {
+        final double lambda0 = lambda0(alpha, beta, f, a, r);
+        final int n = a + r;
+        final double kappa = kappa(alpha, f, r, n, lambda0);
+        final double rho = rho(lambda0, kappa);
+        final double tau = tau(lambda0, kappa);
+        final double logc = alpha*log(beta) - Gamma.logGamma(alpha) + a * log(f) + r * log(1 - f)
+                + (r + alpha - rho) * log(lambda0) + (tau - beta) * lambda0 - n * log(f + (1 - f) * lambda0);
+        return logc + Gamma.logGamma(rho) - rho * log(tau);
+    }
+
+    protected static double lambda0(final double alpha, final double beta, final double f, final int a, final int r) {
```

Comment:
`lambda0` is the mode the the posterior obtained from the Gamma prior with parameters `alpha` and `beta` and the likelihood given by `a`, `r`, and `f`.  As long as you're extracting a method -- which is definitely an improvement -- I would rename the function and give it a javadoc explaining all the parameters (or, at the price of bulkier equations, rename the parameters -- you're choice).